lib/auth/notifications/notificationsv1/service.go

```diff
@@ -307,6 +307,10 @@ func (s *Service) matchGlobalNotification(ctx context.Context, authCtx *authz.Co
 		// Always return true if the matcher is "all."
 		return true
 
+	case *notificationsv1.GlobalNotificationSpec_ByUsers:
+		userList := matcher.ByUsers.GetUsers()
+		return slices.Contains(userList, authCtx.User.GetName())
+
```

Comment:
is there a test for this? or exclude this change?

Reply:
added a test

Reply:
added a test for this matcher